experimental/framework/fluid-static/src/containerCode.ts

```diff
@@ -20,27 +20,48 @@ export interface IFluidStaticDataObjectClass {
     readonly factory: IFluidDataStoreFactory;
 }
 
-export class RootDataObject extends DataObject {
-    protected async initializingFirstTime() { }
+interface RootDataObjectProps {
+    initialObjects: IdToDataObjectCollection;
+}
+
+// eslint-disable-next-line @typescript-eslint/ban-types
+export class RootDataObject extends DataObject<{}, RootDataObjectProps> {
+    protected async initializingFirstTime(props: RootDataObjectProps) {
+        // Create initial objects provided by the developer
+        const initialObjectsP: Promise<void>[] = [];
+        Object.entries(props.initialObjects).forEach(([id, dataObjectClass]) => {
+            const createObject = async () => {
+                const obj = await this.createInternal(dataObjectClass);
+                this.root.set(id, obj.handle);
+            };
+            initialObjectsP.push(createObject());
+        });
+
+        await Promise.all(initialObjectsP);
+    }
 
     protected async hasInitialized() { }
 
-    public async createDataObject<T extends DataObject>(
+    public async createDataObject<T extends IFluidLoadable>(
         dataObjectClass: IFluidStaticDataObjectClass,
         id: string,
     ) {
-        const factory = dataObjectClass.factory;
-        const packagePath = [...this.context.packagePath, factory.type];
-        const router = await this.context.containerRuntime.createDataStore(packagePath);
-        const object = await requestFluidObject<T>(router, "/");
-        this.root.set(id, object.handle);
-        return object;
+        const obj = await this.createInternal(dataObjectClass);
+        this.root.set(id, obj.handle);
+        return obj;
     }
 
-    public async getDataObject<T extends DataObject>(id: string) {
+    public async getDataObject<T extends IFluidLoadable>(id: string) {
         const handle = await this.root.wait<IFluidHandle<T>>(id);
         return handle.get();
     }
+
+    private async createInternal<T extends IFluidLoadable>(dataObjectClass: IFluidStaticDataObjectClass) {
+        const factory = dataObjectClass.factory;
+        const packagePath = [...this.context.packagePath, factory.type];
+        const router = await this.context.containerRuntime.createDataStore(packagePath);
+        return requestFluidObject<T>(router, "/");
+    }
 }
 
 const rootDataStoreId = "rootDOId";
```

Comment:
Just noticing this, should we flag that this is not a valid id? Do we have a known error for that?

Reply:
It'd be a separate issue/PR

Reply:
"rootDOId" points to the id of the RootDataObject. This will be stored as the id of the DataObject on the container. The create path stores new DataObjects on the root map of the RootDataObject so there will be no conflicts.

My next PR will shake things up a good amount but we will not be allows callers to definte the id of DataObjects from the runtime perspective.